src/datanode/src/datanode.rs

```diff
@@ -655,12 +665,25 @@ async fn open_all_regions(
     }
 
     for region_id in open_regions {
-        if open_with_writable
-            && let Err(e) = region_server.set_region_role(region_id, RegionRole::Leader)
-        {
-            error!(
-                e; "failed to convert region {region_id} to leader"
-            );
+        if open_with_writable {
+            let res = region_server.set_region_role(region_id, RegionRole::Leader);
+            match res {
+                Ok(_) => {
+                    // Finalize leadership: persist backfilled metadata.
+                    if let SetRegionRoleStateResponse::InvalidTransition(err) = region_server
+                        .set_region_role_state_gracefully(
+                            region_id,
+                            SettableRegionRoleState::Leader,
+                        )
+                        .await?
+                    {
+                        error!(err; "failed to convert region {region_id} to leader");
+                    }
+                }
+                Err(e) => {
+                    error!(e; "failed to convert region {region_id} to leader");
+                }
+            }
```

Comment:
How about to call `set_region_role_state_gracefully` directly?

Reply:
Tried but follower cannot be transfered to leader via `set_region_role_state_gracefully `.